pkg/storage/etcd/etcd_helper.go

```diff
@@ -154,7 +155,6 @@ type etcdHelper struct {
 	// This depends on etcd's indexes being globally unique across all objects/types. This will
```

Comment:
this comment is concerning... we already allow segmenting by type, which means we can't assume an etcd index is globally unique